dockerfiles/jaeger/agent/Dockerfile

```diff
@@ -0,0 +1,18 @@
+# This Dockerfile tracks jaegertracing/jaeger-agent, but includes more debugging tools and runs as a
+# non-root user. It requires JAEGER_VERSION to be set as an argument to build.
+ARG JAEGER_VERSION
+FROM jaegertracing/jaeger-agent:$JAEGER_VERSION as base
+
+FROM alpine:3.11.5
+RUN apk update
+RUN apk add bash curl
+
+COPY --from=base /etc/ssl/certs/ca-certificates.crt /etc/ssl/certs/ca-certificates.crt
```

Comment:
you could just install ca-certificates in the above line, would seem "safer". Also do we need `bind-tools` @slimsag? this agent does need to resolve the address of the jaeger collector. However, I don't think bind-tools is needed in k8s?

Reply:
Is `bind-tools` necessary for DNS resolution? If so, it may be required for the "send spans to external jaeger-collector" use case (which is documented, but I haven't actually tested). I will check that scenario.

`apk add ca-certificates` is indeed [what is run in the standard jaeger-agent image](https://github.com/jaegertracing/jaeger/blob/master/cmd/agent/Dockerfile#L2). We could run it here instead of copying the value from the standard jaeger-agent image. Benefit would be we might have slightly more up-to-date certs, but at the cost of not as closely tracking the source image. My preference is to track the base image as closely as possible (and update the base image version if an update is required).

Reply:
> Is bind-tools necessary for DNS resolution?

I think at some customers using the deploy-sourcegraph-docker method needed the bind-tools package. I don't think it is needed for k8s clusters, but just wanted to confirm that.

> If so, it may be required for the "send spans to external jaeger-collector" use case (which is documented, but I haven't actually tested). I will check that scenario.

Isn't it also needed to send to the collector in the cluster? IE we do a dns lookup, don't use a hardcoded IP.

> apk add ca-certificates is indeed what is run in the standard jaeger-agent image . We could run it here instead of copying the value from the standard jaeger-agent image. Benefit would be we might have slightly more up-to-date certs, but at the cost of not as closely tracking the source image. My preference is to track the base image as closely as possible (and update the base image version if an update is required).

Yeah sounds good to just keep this line as it is then.

Reply:
> Isn't it also needed to send to the collector in the cluster? IE we do a dns lookup, don't use a hardcoded IP.

I don't think so? At least, the jaeger-agent has no problem talking to the collector in the test cluster I spun up.

Reply:
`bind-tools` is needed for DNS resolution in Docker environments. See https://sourcegraph.com/github.com/sourcegraph/sourcegraph@730f31fb13b8380ed02fd01fe4a8583e43b029d4/-/blob/docker-images/alpine/Dockerfile#L17-23